### PR TITLE
Home Work 2

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: untitled/src/ArraySorter.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class ArraySorter {
+
+	void arraySorter(int[] array1, int[] array2) {
+		int[] complitedArray = new int[array1.length + array2.length];
+		int indexForFirstArray = 0, indexForSecondArray = 0, indexForThirdArray = 0;
```
Comment (SevostyanovGerman) [inline untitled/src/ArraySorter.java]: cs

File: untitled/src/AsciiCharSequence.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Override
+	public char charAt(int index) {
+		return (char) stringReplace[index];
+	}
+
+	@Override
+	public CharSequence subSequence(int start, int end) {
+		byte[] result = new byte[end - start];
+		if (end - start >= 0) System.arraycopy(stringReplace, start, result, start - start, end - start);
```
Comment (SevostyanovGerman) [inline untitled/src/AsciiCharSequence.java]: cs

File: untitled/src/AsciiCharSequence.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		return new AsciiCharSequence(result);
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder result = new StringBuilder(stringReplace.length);
```
Comment (SevostyanovGerman) [inline untitled/src/AsciiCharSequence.java]: simplify

File: untitled/src/ComplexNumber.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//	@Override
+//	public boolean equals(Object o) {
+//		if (this == o) return true;
+//		if (o == null || getClass() != o.getClass()) return false;
+//		ComplexNumber complexNumber = (ComplexNumber) o;
+//		if (Double.compare(complexNumber.re, re) != 0) return false;
+//		return Double.compare(complexNumber.im, im) == 0;
+//	}
+
+	@Override
```
Comment (SevostyanovGerman) [inline untitled/src/ComplexNumber.java:37]: write equals hashcode yourself

File: untitled/src/Palindrome.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Palindrome {
+	void ispPlindrome(String stringForCheck) {
+		String cleanString = stringForCheck.toUpperCase().replaceAll("[\\-+.^:,]", "");
```
Comment (SevostyanovGerman) [inline untitled/src/Palindrome.java]: it is dosent work in stepic

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: untitled/src/AsciiCharSequence.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	@Override
+	public char charAt(int index) {
+		return (char) stringReplace[index];
+	}
+
+	@Override
+	public CharSequence subSequence(int start, int end) {
+		byte[] result = new byte[end - start];
+		if (end - start >= 0) {
+			System.arraycopy(stringReplace, start, result, start - start, end - start);
```
Comment (Sofasmile) [inline untitled/src/AsciiCharSequence.java:26]: better change
start - start -> 0

File: untitled/src/ComentCheck.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
+public class ComentCheck {
+
+	abstract class KeywordAnalyzer implements TextAnalyzer {
```
Comment (Sofasmile) [inline untitled/src/ComentCheck.java]: better create all classes separately

File: untitled/src/Factorial.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+class Factorial {
+
+	void factorialFinder(int number) {
```
Comment (Sofasmile) [inline untitled/src/Factorial.java]: read task and use BigInteger 
https://stepik.org/lesson/12762/step/8?unit=3110

File: untitled/src/Palindrome.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+class Palindrome {
+	public boolean isPalindrome(String stringForCheck) {
+		String regex = "[^A-Za-z0-9]";
+		String cleanString = stringForCheck.replaceAll(regex, "");
+		String revertString = new StringBuilder(cleanString).reverse().toString();
+		if (revertString.equalsIgnoreCase(cleanString)) {
```
Comment (Sofasmile) [inline untitled/src/Palindrome.java]: optional: you can use ternary operator here

File: untitled/src/Palindrome.java (hunk 5 above)
Comment (UnitOmega13) [reply untitled/src/Palindrome.java]: have changed the code for stepik

File: untitled/src/Robot.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
+public static class Robot {
+	int x;
```
Comment (Sofasmile) [inline untitled/src/Robot.java]: add access modifier

File: untitled/src/Robot.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+			y--;
+		}
+		if (direction == Direction.LEFT) {
+			x--;
+		}
+		if (direction == Direction.RIGHT) {
+			x++;
+		}
+	}
+}
```
Comment (Sofasmile) [inline untitled/src/Robot.java]: add blank line

File: untitled/src/RobotMove.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
+class RobotMove {
+
+		public enum Direction {
```
Comment (Sofasmile) [inline untitled/src/RobotMove.java]: better move enum into separate file

**Review by Sofasmile — COMMENTED** [review]
resolve comments

File: untitled/src/RobotMove.java (hunk 12 above)
Comment (UnitOmega13) [reply untitled/src/RobotMove.java]: Fix all from comments and test all on stepik

File: untitled/src/ArraySorter.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+class ArraySorter {
+
+	int[] arraySorter(int[] a1, int[] a2) {
+		int[] complitedArray = new int[a1.length + a2.length];
+		int indexForFirstArray = 0;
+		int indexForSecondArray = 0;
+		int indexForThirdArray = 0;
+		while (indexForFirstArray < a1.length && indexForSecondArray < a2.length) {
+			complitedArray[indexForThirdArray++] = a1[indexForFirstArray] < a2[indexForSecondArray] ? a1[indexForFirstArray++] : a2[indexForSecondArray++];
```
Comment (Sofasmile) [inline untitled/src/ArraySorter.java]: very long line
move part of it at the next line

**Review by Sofasmile — APPROVED** [review]
good, cs

File: untitled/src/TooLongTextAnalyzer.java [hunk 14]
```diff
@@ -0,0 +1,9 @@
+package src;
+
+public abstract class TooLongTextAnalyzer implements TextAnalyzer {
+	private int maxLength;
+
+	public TooLongTextAnalyzer(int threshold) {
+		this.maxLength = threshold;
+	}
+}
```
Comment (Sofasmile) [inline untitled/src/TooLongTextAnalyzer.java]: add a blank line

**Review by Sofasmile — COMMENTED** [review]
good

**Review by Sofasmile — APPROVED** [review]
good

**Pull request closed without merging**